modules/cat/fastq/main.nf

```diff
@@ -8,7 +8,7 @@ process CAT_FASTQ {
         'biocontainers/biocontainers:v1.2.0_cv1' }"
 
     input:
-    tuple val(meta), path(reads)
+    tuple val(meta), path(reads, stageAs: "input*/*")
```

Comment:
I think the linting may be failing because of this syntax 😏 @grst how easy would it be to fix this?

Reply:
I can confirm the checksum mismatch in the test is caused by the reversed order of the fastq files in `cat` command. This is because `read1.sort()` will return different output when the fastq file names are prefixed with subfolder e.g. `input1`. I can update the checksum in `test,yml` if the order of the fastq files to concatenate doesn't matter.

Reply:
@jinmingda I had a play with this locally and it appears that for paired-end data the sorting would break the ordering of the files. For single-end data it wouldn't matter as much but we would need to preserve the read order for paired-end files.

```
cat input1/SRR6357070_1.fastq.gz input3/SRR6357070_1.fastq.gz input5/SRR6357070_1.fastq.gz input7/SRR6357071_1.fastq.gz input9/SRR6357071_1.fastq.gz > WT_REP1_1.merged.fastq.gz
cat input10/SRR6357071_2.fastq.gz input2/SRR6357070_2.fastq.gz input4/SRR6357070_2.fastq.gz input6/SRR6357070_2.fastq.gz input8/SRR6357071_2.fastq.gz > WT_REP1_2.merged.fastq.gz
```

Reply:
Removing the `sort()` as I pushed in https://github.com/nf-core/modules/pull/1107/commits/a9f472a0c6af4e921ec82c5930f48c5b5c557841 will give us the correct ordering now 🎉 :

## Paired-end
```
cat input1/SRR6357070_1.fastq.gz input3/SRR6357070_1.fastq.gz input5/SRR6357070_1.fastq.gz input7/SRR6357071_1.fastq.gz input9/SRR6357071_1.fastq.gz > WT_REP1_1.merged.fastq.gz
cat input2/SRR6357070_2.fastq.gz input4/SRR6357070_2.fastq.gz input6/SRR6357070_2.fastq.gz input8/SRR6357071_2.fastq.gz input10/SRR6357071_2.fastq.gz > WT_REP1_2.merged.fastq.gz
```

## Single-end
```
cat input1/SRR6357074_1.fastq.gz input2/SRR6357075_1.fastq.gz input3/SRR6357075_1.fastq.gz > RAP1_UNINDUCED_REP2.merged.fastq.gz
```

Reply:
@drpatelh thank you for the follow-up review. The changes look good to me. I can't think of any edge case for now. But I am not sure why the linting failed.

Reply:
No worries. The linting is broken because we haven't used that particular `stageAs:` syntax yet and so it is raising a parsing error in Python.